### ISSUE TITLE
Invert button order

### ISSUE THREAD

**Issue by hmlendea** [franglais125/no-title-bar#28]
Hello.

Can you add a toggle to invert the order of the buttons?
It feels really weird to have [Minimise, Maximise, Close] when having them in the left hand part of the panel. When inverted they should be: [Close, Maximise, Minimise]

**Comment by franglais125** [comment]
Hi! Thanks for the suggestion, I'll try to tackle this when I have some time.

In the meantime, I don't really understand the logic.

1. What do you mean by "when inverted"? And in what way does it feel weird?
2. I know that this is perhaps not exactly what you are looking for, but you can play with the order by using:
`gsettings set org.gnome.desktop.wm.preferences button-layout ':close,maximize,minimize'`. The order you use (and the `:`) sets the order and position of the buttons for the Shell, and for this extension.

**Comment by hmlendea** [comment]
I tried the gsetting. It changed the ones on title bars but not the ones from the extension. To be honest, I haven't restarted the system since. I will try it again today.

**Comment by franglais125** [comment]
@hmlendea any luck?

**Comment by hmlendea** [comment]
Yes, sorry for not reporting earlier. Yes, it works :) However I have another issue as the position of the buttons gets reset after restarting the shell. It keeps putting itself at the left end of the bar, right in the corner, when in reality I configured it to stay at the right of the AppMenu.

**Comment by echonaut** [comment]
> What do you mean by "when inverted"? And in what way does it feel weird?

I believe what he's referring to is the fact that the left-to-right order of the buttons changes when you change the window buttons to "Left" in the Gnome Tweak Tool. It doesn't *invert* the buttons, which is why it might be confusing, but it reorders them, to what OSX uses (you can see it in [this screenshot](https://cdn.onemorething.nl/uploads/2017/09/macOS-High-Sierra-Storage.jpg)).

TLDR: When GNOME is set to put window caption buttons on the right, the order is: [ _ □ x ], which is what this extension ALWAYS uses, despite the position on the top bar.

When GNOME is set to put the window caption buttons on the LEFT, the order is: [ X _ □ ], which isn't inverted but it IS reordered. 

I think it feels "weird" in that OSX kind of standardized the order of caption buttons on the left side of the window, to the extend that GNOME copied OSX for that particular bit.

**Comment by echonaut** [comment]
FYI regarding this issue: if you're using this extension and you change the setting for buttons on left or right side, WHILE THE EXTENSION IS ENABLED, it'll reflect the previous setting. You have to disable and re-enable the extension to make it respect the change.

**Comment by hmlendea** [comment]
@echonaut I did. And the order of the buttons is fine. Now I have other problems, unfortunately :(

**Comment by franglais125** [comment]
Hey guys, thanks for the feedback on this. I'll try to see what I can do about it when I get some time. As I have expressed elsewhere, patches are welcome if anyone is wiling to look into this.
